connexion/lifecycle.py

```diff
@@ -34,4 +34,21 @@ def __init__(self,
         self.mimetype = mimetype
         self.content_type = content_type
         self.body = body
-        self.headers = headers or {}
+        self.headers = ConnexionHeaders(headers)
+
+
+class ConnexionHeaders(object):
```

Comment:
We use it for [authorization](https://github.com/zalando/connexion/blob/53908d30e566ccb81d73d3e27b461f407a8a8d31/connexion/decorators/security.py#L56), for example. It's a legitimate use case if it only makes sense to have one value for a certain header, and that's why frameworks, including flask support it.